lib/ansible/executor/process/worker.py

```diff
@@ -127,12 +127,16 @@ def run(self):
         finally:
             # This is a hack, pure and simple, to work around a potential deadlock
             # in ``multiprocessing.Process`` when flushing stdout/stderr during process
-            # shutdown. We have various ``Display`` calls that may fire from a fork
-            # so we cannot do this early. Instead, this happens at the very end
-            # to avoid that deadlock, by simply side stepping it. This should not be
-            # treated as a long term fix.
-            # TODO: Evaluate overhauling ``Display`` to not write directly to stdout
-            # and evaluate migrating away from the ``fork`` multiprocessing start method.
+            # shutdown.
+            #
+            # We should no longer have a problem with ``Display``, as it now proxies over
+            # the queue from a fork. However, to avoid any issues with plugins that may
+            # be doing their own printing, this has been kept.
+            #
+            # This happens at the very end to avoid that deadlock, by simply side
+            # stepping it. This should not be treated as a long term fix.
```

Comment:
should we 'redefine print' on forks to avoid this?

Reply:
that is a much bigger change, it might still be worth redefining print until we get there